tests/unittests/tests-core/tests-core-mbox.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2022 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <limits.h>
+
+#include "embUnit.h"
+
+#include "mbox.h"
+
+#include "tests-core.h"
+
+#define QUEUE_SIZE  8
+
+static unsigned gen_val(unsigned i) {
+    return i + 1337;
+}
+
+static void test_mbox_put_get(void)
```

Comment:
Why not be more verbose? For example, `test_mbox_reject_message_if_queue_is_full()` [or adding comments to the current code] gives a good hint on what kind of behaviour to expect from mbox.

Reply:
Note that splitting it into smaller functions would have multiple downsides:

- code size (which is indeed an issue for the unit tests)
- setting up the mbox in the situation to test (e.g. try_put failing on a full mbox) assumes that mox_try_put does actually work at least to some degree correctly. We may end up either having the setup also being carefully checked, or misleading users to assume that e.g. try_put succeeds on a full mbox, while maybe the mbox just wasn't full due to another issue.
- more overhead to read (right now the function is neither too long nor too complex to grasp in its entirety with a quick look, so there is not yet anything to gain from splitting it)

I could add some comments, though